src/EditorFeatures/Core.Wpf/Interactive/InteractiveDocumentNavigationService.cs

```diff
@@ -39,7 +38,7 @@ public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSp
             var textView = interactiveWorkspace.Window.TextView;
             var document = interactiveWorkspace.CurrentSolution.GetDocument(documentId);
 
-            var textSnapshot = document.GetTextSynchronously(CancellationToken.None).FindCorrespondingEditorTextSnapshot();
+            var textSnapshot = document.GetTextSynchronously(cancellationToken).FindCorrespondingEditorTextSnapshot();
```

Comment:
an example of a problematic area.  If this op took time (which it can since it may involve blocking IO), you are effectively completely blocked in VS, even if you hit hte cancel button on a CWT.